packages/vaadin-element-mixin/test/templates.test.js

```diff
@@ -0,0 +1,69 @@
+import sinon from 'sinon';
+import { fixtureSync } from '@vaadin/testing-helpers';
+import { expect } from '@esm-bundle/chai';
+import { PolymerElement } from '@polymer/polymer/polymer-element.js';
+
+import { processTemplates } from '../templates.js';
+
+class ComponentElement extends PolymerElement {
+  static get is() {
+    return 'component-element';
+  }
+}
+
+customElements.define(ComponentElement.is, ComponentElement);
+
+describe('process templates', () => {
+  beforeEach(() => {
+    sinon.stub(console, 'warn');
+  });
+
+  afterEach(() => {
+    console.warn.restore();
+  });
+
+  describe('with template renderer', () => {
+    beforeEach(() => {
+      window.Vaadin = window.Vaadin || {};
+      window.Vaadin.templateRendererCallback = sinon.spy();
+    });
+
+    afterEach(() => {
+      window.Vaadin.templateRendererCallback = undefined;
+    });
+
+    it('should invoke the template renderer callback', () => {
+      const component = fixtureSync(`
+        <component-element>
+          <template></template>
+        </component-element>
+      `);
+
+      processTemplates(component);
```

Comment:
Rather than calling this manually in the test, shouldn't there be
```js
ready() {
  super.ready();
  processTemplates(this);
}
```

in `ComponentElement` instead?

Reply:
Good point! Fixed.